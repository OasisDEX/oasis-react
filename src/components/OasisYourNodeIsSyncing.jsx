import React, { PureComponent } from "react";
import { PropTypes } from "prop-types";
// import ImmutablePropTypes from 'react-immutable-proptypes';

import CSSModules from "react-css-modules";
import styles from "./OasisYourNodeIsSyncing.scss";

import OasisLogo from "./../assets/logo-oasis-hover.png";
import loading from "./../assets/od-icons/icon-syncing.svg";
import OasisFooterWrapper from "../containers/OasisFooter";
import web3 from "../bootstrap/web3";
import moment from "moment";
import { KOVAN_NET_ID, MAIN_NET_ID } from "../constants";

const propTypes = PropTypes && {
  latestBlock: PropTypes.object,
  networkId: PropTypes.string
};
const defaultProps = {};

class OasisYourNodeIsSyncing extends PureComponent {
  getNetworkName() {
    const { networkId } = this.props;

    switch (networkId) {
      case MAIN_NET_ID:
        return "Homestead";
      case KOVAN_NET_ID:
        return "Kovan";
    }
  }

  formatBlockTime() {
    const { latestBlock: { timestamp } } = this.props;
    return moment.unix(timestamp).format("llll");
  }

  render() {
    const { latestBlock: { number } = {} } = this.props;
    return (
      <section className={styles.LockedAccountSection}>
        <div>
          <div style={{ textAlign: "center" }}>
            <img
              src={loading}
              style={{ width: "400px", position: "relative" }}
              alt=""
            />
          </div>
          <div className={styles.LogoContainer}>
            <img
              className={styles.OasisLogo}
              alt="Oasis Logo"
              style={{ position: "relative", bottom: "250px" }}
              src={OasisLogo}
            />
          </div>
        </div>
        <hr className={styles.Separator} />
        <h1 className={styles.Heading}>Your node this syncing</h1>
        <hr className={styles.Separator} />
        <p className={styles.Message}>
          <span
            style={{
              background: "white",
              color: "#6f727d",
              border: "1px solid #6f727d",
              padding: "10px",
              borderRadius: "10px 0 0 10px"
            }}
          >
            Network
          </span>
          <span
            style={{
              background: "#6f727d",
              color: "white",
              padding: "10px",
              borderRadius: "0 10px 10px 0"
            }}
          >
            {this.getNetworkName()}
          </span>
        </p>
        <p className={styles.Message}>
          <span
            style={{
              background: "white",
              color: "#6f727d",
              border: "1px solid #6f727d",
              padding: "10px",
              borderRadius: "10px 0 0 10px"
            }}
          >
            Current block
          </span>
          <span
            style={{
              background: "#6f727d",
              color: "white",
              padding: "10px",
              borderRadius: "0 10px 10px 0"
            }}
          >
            {web3.toBigNumber(number).toFormat()} @ {this.formatBlockTime()}
          </span>
        </p>
        <hr className={styles.Separator} />
        <OasisFooterWrapper />
      </section>
    );
  }
}

OasisYourNodeIsSyncing.displayName = "OasisYourNodeIsSyncing";
OasisYourNodeIsSyncing.propTypes = propTypes;
OasisYourNodeIsSyncing.defaultProps = defaultProps;
export default CSSModules(OasisYourNodeIsSyncing, styles, {
  allowMultiple: true
});
