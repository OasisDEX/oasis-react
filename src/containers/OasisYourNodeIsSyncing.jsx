import React, { PureComponent } from "react";
import { PropTypes } from "prop-types";
// import ImmutablePropTypes from 'react-immutable-proptypes';

import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import OasisYourNodeIsSyncing from "../components/OasisYourNodeIsSyncing";
import network from "../store/selectors/network";

const propTypes = PropTypes && {
  actions: PropTypes.object.isRequired
};

export class OasisYourNodeIsSyncingWrapper extends PureComponent {
  render() {
    const { latestBlock, networkId } = this.props;
    return (
      <OasisYourNodeIsSyncing latestBlock={latestBlock} networkId={networkId} />
    );
  }
}

export function mapStateToProps(state) {
  return {
    latestBlock: network.latestBlock(state),
    networkId: network.activeNetworkId(state)
  };
}
export function mapDispatchToProps(dispatch) {
  const actions = {};
  return { actions: bindActionCreators(actions, dispatch) };
}

OasisYourNodeIsSyncingWrapper.propTypes = propTypes;
OasisYourNodeIsSyncingWrapper.displayName = "OasisYourNodeIsSyncing";
export default connect(mapStateToProps, mapDispatchToProps)(
  OasisYourNodeIsSyncingWrapper
);
