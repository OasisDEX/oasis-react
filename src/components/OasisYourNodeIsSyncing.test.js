/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisYourNodeIsSyncing from './OasisYourNodeIsSyncing';


describe('(Component) OasisYourNodeIsSyncing', () => {
  it('should render', () => {
    const props = {
      latestBlock: {}
    };
    const wrapper = shallow(
      <OasisYourNodeIsSyncing {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
