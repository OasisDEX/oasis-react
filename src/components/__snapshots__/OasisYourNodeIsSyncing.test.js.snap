// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisYourNodeIsSyncing should render 1`] = `
<section>
  <div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <img
        alt=""
        src="icon-syncing.svg"
        style={
          Object {
            "position": "relative",
            "width": "400px",
          }
        }
      />
    </div>
    <div>
      <img
        alt="Oasis Logo"
        src="logo-oasis-hover.png"
        style={
          Object {
            "bottom": "250px",
            "position": "relative",
          }
        }
      />
    </div>
  </div>
  <hr />
  <h1>
    Your node this syncing
  </h1>
  <hr />
  <p>
    <span
      style={
        Object {
          "background": "white",
          "border": "1px solid #6f727d",
          "borderRadius": "10px 0 0 10px",
          "color": "#6f727d",
          "padding": "10px",
        }
      }
    >
      Network
    </span>
    <span
      style={
        Object {
          "background": "#6f727d",
          "borderRadius": "0 10px 10px 0",
          "color": "white",
          "padding": "10px",
        }
      }
    />
  </p>
  <p>
    <span
      style={
        Object {
          "background": "white",
          "border": "1px solid #6f727d",
          "borderRadius": "10px 0 0 10px",
          "color": "#6f727d",
          "padding": "10px",
        }
      }
    >
      Current block
    </span>
    <span
      style={
        Object {
          "background": "#6f727d",
          "borderRadius": "0 10px 10px 0",
          "color": "white",
          "padding": "10px",
        }
      }
    >
      0
       @ 
      Invalid date
    </span>
  </p>
  <hr />
  <Connect(OasisFooterWrapper) />
</section>
`;
